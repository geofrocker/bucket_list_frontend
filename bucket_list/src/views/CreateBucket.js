import React, {Component} from 'react';
import Header from '../components/Header';
import Sidebar from '../components/Sidebar';
import axios from 'axios'

class CreateBucket extends Component{
    constructor(props){
        super(props);
        this.state = {bucket_name:'', description:'', category:'', isAuthorized:false};
        this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
        this.handleNameChange = this.handleNameChange.bind(this);
        this.handleCategoryChange = this.handleCategoryChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    // componentDidMount(){
    //     axios({
    //         url: 'https://ridge-bucket-list-api.herokuapp.com/api/v1/bucketlists/',
    //         method: "GET"
    //     }).then(({data}) =>{
    //         console.log(data);
    //         this.setState({array: data})
    //     })
    //
    // }

    handleNameChange(event){
        this.setState({bucket_name: event.target.value})
    }

    handleDescriptionChange(event){
        this.setState({description: event.target.value})
    }

    handleCategoryChange(event){
        this.setState({category: event.target.value})
    }

    handleSubmit(event){
        let data = {
            bucket_name:this.state.bucket_name,
            description: this.state.description,
            category: this.state.category
        };
        
        axios({
            url: 'http://ridge-bucket-list-api.herokuapp.com/api/v1/bucketlists/',
            method: 'POST',
            datatype: "json",
            data: data,
            
        }).then(function (response) {
            alert(response.response.data)
        }). catch(function (xhr) {
            if ("response" in xhr){
                alert(xhr.response.data.error);
            }
        })
    }

    render(){
        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content item-editor-page">
                    <div className="title-block">
                        <h3 className="title"> Add new Bucket
                            <span className="sparkline bar"></span>
                        </h3>
                    </div>

                    <form name="item" method="post" onsubmit={this.handleSubmit}>
                        <div className="card card-block">
                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Name:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="bucket_name" value={this.state.bucket_name} onChange={this.handleNameChange} className="form-control boxed" placeholder="" />
                                </div>
                            </div>

                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Description:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="description" value={this.state.description} onChange={this.handleDescriptionChange} className="form-control boxed" placeholder="" />
                                </div>
                            </div>

                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Category:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="category" value={this.state.category} onChange={this.handleCategoryChange} className="form-control boxed" placeholder="Eg. Travel" />
                                </div>
                            </div>

                            <div className="col-sm-10 col-sm-offset-2">
                                <button type="submit" className="btn btn-primary">
                                    Add
                                </button>
                                <button type="submit" className="btn btn-primary">
                                    Add & New
                                </button>
                            </div>
                        </div>
                    </form>
                </article>
            </div>

        )
    }
}
export default CreateBucket;