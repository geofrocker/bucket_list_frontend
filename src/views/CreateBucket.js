import React, {Component} from 'react';
import Header from '../components/Header';
import Sidebar from '../components/Sidebar';
import {Redirect} from 'react-router-dom';
import axios from 'axios';
import swal from 'sweetalert';

class CreateBucket extends Component{
    constructor(props){
        super(props);
        this.state = {bucket_name:'', description:'', category:'', token: '',isAuthorized:false, login_redirect:false};
        this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
        this.handleNameChange = this.handleNameChange.bind(this);
        this.handleCategoryChange = this.handleCategoryChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount() {
        axios({
            url: 'http://127.0.0.1:5000/api/v1/callback',
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then(()=>{
                this.setState({isAuthorized:true});
        })
            .catch(() =>{
                this.setState({login_redirect:true});
            });
    }

    handleNameChange(event){
        this.setState({bucket_name: event.target.value})
    }

    handleDescriptionChange(event){
        this.setState({description: event.target.value})
    }

    handleCategoryChange(event){
        this.setState({category: event.target.value})
    }


    handleSubmit(event){
        event.preventDefault();
        let data = {
            bucket_name:this.state.bucket_name,
            description: this.state.description,
            category: this.state.category
        };
        
        axios({
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/',
            method: 'POST',
            datatype: "json",
            data: data,
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
            
        })
            .then(()=>{
                this.setState({redirect:true});
        })
            .catch((xhr) =>{
                swal("Error!", xhr.response.data.error, "error");
                window.localStorage.setItem('isLoggedIn', false)
        });
    }

    render(){

        if(this.state.login_redirect){
            return(<Redirect to={'/login/'}/>)
        }

        if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">

                    </article>
                </div>
            )
        }
        else if (this.state.redirect === true){
            return <Redirect to="/bucketlists/view"/>
        }
        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content item-editor-page">
                    <div className="title-block">
                        <h3 className="title"> Add new Bucket
                            <span className="sparkline bar"></span>
                        </h3>
                    </div>

                    <form method="post" id="create_bucket_form" onSubmit={this.handleSubmit}>
                        <div className="card card-block">
                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Name:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="bucket_name" value={this.state.bucket_name} onChange={this.handleNameChange} className="form-control boxed" placeholder="" />
                                </div>
                            </div>

                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Description:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="description" value={this.state.description} onChange={this.handleDescriptionChange} className="form-control boxed" placeholder="" />
                                </div>
                            </div>

                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Category:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="category" value={this.state.category} onChange={this.handleCategoryChange} className="form-control boxed" placeholder="Eg. Travel" />
                                </div>
                            </div>

                            <div className="col-sm-10 col-sm-offset-2">
                                <button type="submit" className="btn btn-primary">
                                    Add
                                </button>
                            </div>
                        </div>
                    </form>
                </article>
            </div>

        )
    }
}
export default CreateBucket;