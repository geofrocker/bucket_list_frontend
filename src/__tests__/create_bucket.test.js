import React from 'react';
import CreateBucket from '../components/CreateBucket';

import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

describe('<CreateBucket />', ()=>{
    beforeEach(() => {
        global.localStorage = {
            token: 'someToken',
            getItem: function () {
                return 'someToken'
            }
        };

    });

    it('contains create bucket form', function () {
        const wrapper = mount(
            <MemoryRouter>
                <CreateBucket/>
            </MemoryRouter>);

        const preventDefault = jest.fn();
        wrapper.setState({bucket_name:'test_bucket', description:"test description", isAuthorized:true});

        setTimeout(() => {
            expect(wrapper.find("#create_bucket_form").length === 1);
            wrapper.find("#create_bucket_form").simulate('submit', {preventDefault})

        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();


    });

    it('should change values in the form', function () {
        const wrapper = mount(
            <MemoryRouter>
                <CreateBucket/>
            </MemoryRouter>);
        setTimeout(() => {
            wrapper.find("bucket_name").simulate('change', {target: {value: 'test bucket name'}});
            wrapper.find("description").simulate('change', {target: {value:'test description'}});
            wrapper.find("category").simulate('change', {target: {value:'test category'}});
        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();
    })



});

