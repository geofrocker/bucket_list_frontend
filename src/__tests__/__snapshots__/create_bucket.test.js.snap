// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateBucket /> contains create bucket form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "7vnvyi",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "7vnvyi",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CreateBucket>
      <div>
        <article
          className="content item-editor-page"
        />
      </div>
    </CreateBucket>
  </Router>
</MemoryRouter>
`;

exports[`<CreateBucket /> should change values in the form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "o4avlg",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "o4avlg",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CreateBucket>
      <div>
        <article
          className="content item-editor-page"
        />
      </div>
    </CreateBucket>
  </Router>
</MemoryRouter>
`;
